exchanges/graphcache/src/offlineExchange.ts

```diff
@@ -134,21 +134,23 @@ export const offlineExchange = <C extends Partial<CacheExchangeOpts>>(
       );
     };
 
-    storage.onOnline(flushQueue);
-    storage.readMetadata().then(mutations => {
-      if (mutations) {
-        for (let i = 0; i < mutations.length; i++) {
-          failedQueue.push(
-            client.createRequestOperation(
-              'mutation',
-              createRequest(mutations[i].query, mutations[i].variables)
-            )
-          );
-        }
+    storage
+      .readMetadata()
+      .then(mutations => {
+        if (mutations) {
+          for (let i = 0; i < mutations.length; i++) {
+            failedQueue.push(
+              client.createRequestOperation(
+                'mutation',
+                createRequest(mutations[i].query, mutations[i].variables)
+              )
+            );
+          }
 
-        flushQueue();
-      }
-    });
+          flushQueue();
+        }
+      })
+      .finally(() => storage?.onOnline?.(flushQueue));
```

Comment:
```suggestion
      .finally(() => storage.onOnline!(flushQueue));
```

Reply:
The check is redundant as we're already checking for `storage.onOnline` prior, and this is just TypeScript checking in a new scope, so we can guarantee that this is available